process/block/preprocess/sovereignChainTransactions.go

```diff
@@ -204,20 +199,31 @@ func (sct *sovereignChainTransactions) isTransactionEligibleForExecution(tx *tra
 	}
 
 	txFee := sct.economicsFee.ComputeTxFee(tx)
-	if accntInfo.balance.Cmp(txFee) < 0 {
-		log.Trace("sovereignChainTransactions.isTransactionEligibleForExecution", "error", process.ErrInsufficientFee,
-			"account balance", accntInfo.balance.String(),
-			"fee needed", txFee.String())
-		return process.ErrInsufficientFee, false
+	if common.IsRelayedTxV3(tx) && !bytes.Equal(tx.GetSndAddr(), tx.GetRelayerAddr()) {
+		relayerAccount, relayerAccntInfo, errGetRelayer := sct.getRelayerAccountInfo(tx)
+		if check.IfNil(relayerAccount) {
+			log.Debug("sovereignChainTransactions.getRelayerAccountInfo: GetRelayerAccount", "error", errGetRelayer)
+			return errGetRelayer, false
+		}
+
+		errFee := sct.validateAndSubtractTxFee(relayerAccntInfo, txFee)
+		if errFee != nil {
+			return errFee, false
+		}
+		sct.accntsTracker.setAccountInfo(tx.GetRelayerAddr(), relayerAccntInfo)
```

Comment:
In this case, if you have a relayed tx, you execute all code from L221-232. Is that even needed in case of relayed tx since you don't need to modify balance and save `accntInfo`? (or is it required to save in tracker the sender?)

I think the code could be simplified and made more efficient iif saving that account is not required

Reply:
For relayedv3, the tx fee is taken from relayerAddress and the tx value is taken from sender. So, we need to track sender balance always.